### ISSUE TITLE
sky/mojo repository issues

### ISSUE THREAD

**Issue by gauthamt** [domokit/mojo#475]
I've been unable to get versions of my local mojo/sky repositories working with sysui. Tony thinks its a version skew issue.

Here are some flavors of the errors I see (package-root does not seem to be set)
~/sysui/src/theone/android$ flutter run_mojo --mojo-path=../../../../mojo/src --mojo-release --verbose --checked
Unhandled exception:
Uncaught Error: Invalid argument(s): join(null, "sky_tools", "cache"): part 0 was null, but part 1 was not.
Stack Trace:
#0      _validateArgList (package:path/src/context.dart:572)
#1      Context.join (package:path/src/context.dart:210)
#2      join (package:path/path.dart:225)
#3      ArtifactStore._cacheDir.<_cacheDir_async_body> (package:sky_tools/src/artifacts.dart:53)
#4      Future.Future.microtask.<anonymous closure> (dart:async/future.dart:144)
#5      _microtaskLoop (dart:async/schedule_microtask.dart:43)
#6      _microtaskLoopEntry (dart:async/schedule_microtask.dart:52)
#7      _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:96)
#8      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:149)

#0      _rootHandleUncaughtError.<anonymous closure> (dart:async/zone.dart:895)
#1      _microtaskLoop (dart:async/schedule_microtask.dart:43)
#2      _microtaskLoopEntry (dart:async/schedule_microtask.dart:52)
#3      _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:96)
#4      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:149)

:~/sysui/src/theone/android$ /usr/local/google/home/gauthamt/mojo/src/mojo/devtools/common/mojo_run http://app/app.flx --map-origin=http://app/=/usr/local/google/home/gauthamt/mojo/src/examples/dart/camera_roll --android --debug --embed http://app/app.flx --url-mappings=mojo:sky_viewer=https://core.mojoapps.io/sky/sky_viewer.mojo  --verbose
[INFO:network_fetcher.cc(322)] Caching mojo app http://127.0.0.1:31840/icu_data.mojo at /data/data/org.chromium.mojo.shell/app_home/.mojo_url_response_disk_cache/cache/8E1112D1D0DEF88D9F0525105C8056E174722620A471F400B9D11968F768C7C2/8E1112D1D0DEF88D9F0525105C8056E174722620A471F400B9D11968F768C7C2
[1015/162756:ERROR:dart_error.cc(16)] Wrong script snapshot version, expected 'b5452bde33ce72a32401ef7e3cb2ec69' found '515526011f1e746196a748032b93da32'
[1015/162756:ERROR:dart_error.cc(16)] Unhandled exception:
The null object does not have a method 'main'.
NoSuchMethodError: method not found: 'main'
Receiver: null
Arguments: []
#0      Object._noSuchMethod (dart:core-patch/object_patch.dart:42)
#1      Object.noSuchMethod (dart:core-patch/object_patch.dart:45)

**Comment by jason-simmons** [comment]
(I'm new to Mojo but have done some work on Sky)

The first error appears to be a regression in sky_tools.  ArtifactStore.packageRoot is only being set if someone reads the FlutterCommandRunner.buildConfigurations property, and that isn't happening during the run_mojo command.

I was able to work around this by reverting to an earlier commit of sky_tools (d96f5fe870675b4ead9ef853fb89cdf89b684292).

I believe the second error is a Dart version mismatch between the snapshot in the .flx and the sky_viewer.mojo.

I was able to get this to work by using versions of those files that came from my local build of Sky:
  ./out/Debug/mojo_shell file://$SKY_SRC/out/Debug/gen/examples/mine_digger/mine_digger.flx --url-mappings=mojo:sky_viewer=file://$SKY_SRC/out/Debug/sky_viewer.mojo
